### ISSUE TITLE
[Question] Any ideas on enabling DDP with densification/pruning on?

### ISSUE THREAD

**Issue by seigeweapon** [yzslab/gaussian-splatting-lightning#28]
Hello there, great work!

While trying out the multi-gpu training feature, I found the limitation of only allowing DDP with fixed number of gaussians really hurting. A good training always seeks to minimize number of gaussian (smaller output) and maximizing image quality (put more point where needed), so stopping densification/pruning on a very early stage must hurt the quality very much.

May I ask what's the reason of this limitation? Any idea how to solve it? Thanks.

**Comment by yzslab** [comment]
The trainable parameters will change after densification/pruning. But the DDP of PyTorch doesn't follow this changes, so the new Gaussians will not be able to get gradients from other GPUs.
I seldom use DDP, so this has not been fixed yet.
By default the densification/pruning will stop at 15,000 steps, whether you enable DDP or not.
